osu.Game/Online/Leaderboards/LeaderboardErrorState.cs

```diff
@@ -3,10 +3,9 @@
 
 namespace osu.Game.Online.Leaderboards
 {
-    public enum PlaceholderState
+    public enum LeaderboardErrorState
     {
-        Successful,
-        Retrieving,
```

Comment:
not super sure about removing `Retrieving` here - you could fold all of the loading spinner logic into the leaderboard state management rather easily, i think (on every state change, show the spinner if state is `Retrieving`, otherwise hide it - would also fix the bug mentioned above).

Reply:
Main reason for removing it was scoping the naming of this enum. I really didn't like the previous name, and also wanted to make it exclusively for error states if possible.

Reply:
I think that in this particular case there's more value in having the enum represent the general state of the entire control (including the intermediate loading state) than just the end result of a score fetch, but see how you go in fixing the loading spinner visibility thing I guess. Maybe it can be done cleanly without - the main problem I see there with how things are structured right now is that `NoError` is going to be ambiguous (because it can either mean "scores have been fetched OK", or "scores are being fetched right now").

Reply:
What would you call the enum if `Retrieving` was added back?

Reply:
The entire thing you mean? Probably something like `LeaderboardDisplayState` or just even `LeaderboardState`.

Reply:
I'll give it a try.　